instructure/canvas-ios
# Fix media in offline rich content

### What's new?
- URLs in rich content pointing to local offline content were modified to be referenced via relative urls instead of absolute ones.
- I simplified the `HTMLDownloadInteractor` entity to be easier to understand what it does.

refs: [MBL-18070](https://instructure.atlassian.net/browse/MBL-18070)
affects: Student
release note: Fixed images and Studio videos not being available for offline mode after an app update.

test plan:
- Check the ticket for a detailed description on the issue.
- Sync rich content with images and studio videos.
- Go offline and go to the rich content.
- They should appear just as expected.
- Check the offline html content with safari or with a breakpoint in CoreWebView.loadContent (look for the rawHtmlValue variable).
- Urls should be all relative without the application's UUID container ID in it.
- Check out the branch and install the app into the same phone to simulate an app update.
- Start the app in offline mode, rich content should appear just as before.
- Note: we can't test this with other builds because the base url for the offline webview have changed and other builds point to a different one.

## Checklist
